lib/kitchen/transport/winrm.rb

```diff
@@ -452,8 +477,29 @@ def load_needed_dependencies! # rubocop:disable Metrics/MethodLength
       # Load WinRM::Transport code.
       #
       # @api private
+      def host_os_windows?
+        case RbConfig::CONFIG["host_os"]
+        when /darwin/
+          false
+        when /mswin|msys|mingw|cygwin|bccwin|wince|emc/
+          true
+        when /linux/
+          false
+        else
+          false
+        end
+      end
+
+      def sspi_enabled?
+        enabled = host_os_windows? &&
+          config[:winrm_transport] == :sspinegotiate
+        logger.debug("Checking for Negotiate Auth: Enabled = #{enabled}")
+        enabled
+      end
+
       def load_winrm_transport!
         silence_warnings { require "winrm" }
+        silence_warnings { require "winrm-s" } if sspi_enabled?
```

Comment:
I think we're going to need a little load/rescue code for this gem, similar in spirit to [#load_needed_dependencies!](https://github.com/test-kitchen/test-kitchen/blob/b73b8fbccaf84242efb3c05502e36557c963c83a/lib/kitchen/transport/winrm.rb#L426-L450) to handle the case where we want the gem, try to require it, but the gem can't be activated (primarily this for outside ChefDK, but still a valid failure case). I was trying to see where best to handle this, and am currently thinking a short method with the require and `rescue LoadError` blocks? Happy to chat more about this one--I think it's the first case we have of augmenting behavior with an addition gem. Can't think of another place to look for stealing ideas!